crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -61,6 +63,8 @@ module sui::staking_pool {
         pending_total_sui_withdraw: u64,
         /// Pending pool token withdrawn during the current epoch, emptied at epoch boundaries.
         pending_pool_token_withdraw: u64,
+        /// Any extra fields that's not defined statically.
+        extra_fields: Bag,
```

Comment:
Do we still want to add this to types with `key` (given that the `id` already gives us an implicit bag)?

Reply:
Maybe not. Although StakingPool doesn't actually need id. It's probably due to historical reasons.
I think Bag provides a nice abstraction which is it has size, and if you drop with non-zero size it will error (I don't know if general dynamic field would do this today)

Reply:
Just learned that we still use staking pool id as unique identifier for inactive pools. So we probably still need it.
I don't have a strong opinion on whether to use bag here or just direct dynamic field

Reply:
Keeping a bag here sounds fine to me and consistent with what we are doing for other types